### PR TITLE
updates to CONTRIBUTING.adoc

### PR DESCRIPTION
**Pull request by stevenhartley**
Following change elaborates on how to handle code reviews in pull requests so that PRs and reviews are done fairly and reviewers comments are taken into consideration.

### CODE REVIEW

File: CONTRIBUTING.adoc [hunk 1]
```diff
@@ -50,15 +50,23 @@ If the project is _not_ intended to be used for production:
 * *MUST* clearly labeled in your repos README.adoc as `NOT INTENDED FOR PRODUCTION USE`
 
 
+=== Code Reviews
+Code reviews *MUST* be performed by at least one committer before a pull request can be merged.
 
-=== Code Contributions
-Contribution pull requests *MUST* contain the following:
+Reviewers *SHOULD* make suggestions in lieu of just comments, for issues found in pull request following github's guidelines https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request[here]
```
Comment (sophokles73) [inline CONTRIBUTING.adoc:56]: ... found in a pull request following https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request[GitHub's guidelines].

File: CONTRIBUTING.adoc [hunk 2]
```diff
@@ -50,15 +50,23 @@ If the project is _not_ intended to be used for production:
...
 
-* *Commit Description:* Properly formatted subject for the commit including a detailed description of the change. the subject line is pulled into the automatic release generation
+
+=== Pull Requests
+Pull requests *MUST* contain the following:
```
Comment (sophokles73) [inline CONTRIBUTING.adoc:60]: I am not sure if I understand correctly what you want to achieve here. IMHO the things you describe below describe requirements for a *commit* of a PR. Or do you want to define some requirements for the PR comment in GitHub as well? If not, then IMHO we should rephrase this to make it clear that we are talking about *each individual commit* of a PR.

File: CONTRIBUTING.adoc [hunk 3]
```diff
@@ -50,15 +50,23 @@ If the project is _not_ intended to be used for production:
...
+
+* *Commit Description:* Properly formatted subject for the commit including a detailed description of the change. The subject line is pulled into the automatic release generation
 * *GitHub Issue:* Commits for adding new features/functions *SHOULD* have an associated github issue
 * *Documentation:* Interfaces must be properly documented with functional hyperlinks
 * *Test Coverage:* Code *MUST* have associated test coverage 
 
+NOTE: Reviewers comments/suggestions *MUST* be acknowledged by either accepting, rejecting (with explanation), or filing a separate issue, linking the new issue to the pull request, prior to merging the pull request.
```
Comment (sophokles73) [inline CONTRIBUTING.adoc:67]: > Reviewers' comments/suggestions ...

Is this another way of saying that a PR **MUST NOT** be merged before all comments/suggestions have been addressed/resolved?

**Review by sophokles73 — COMMENTED** [review]
In the up libraries diagram, I wonder what the composition relation betwee UPClient Library and uEs is supposed to indicate?

**Comment by stevenhartley** [comment]
> In the up libraries diagram, I wonder what the composition relation betwee UPClient Library and uEs is supposed to indicate?

most upClient libraries have one boundary object that implements the interfaces and you would have-a instance of that boundary object but I didn't feel it is necessary to enforce that on all implementations.

File: CONTRIBUTING.adoc (hunk 3 above)
Comment (stevenhartley) [reply CONTRIBUTING.adoc]: @sophokles73 , yes. I rephrased the section.

File: CONTRIBUTING.adoc (hunk 2 above)
Comment (stevenhartley) [reply CONTRIBUTING.adoc]: @sophokles73 I rephrased the section so it is clearer, thanks for the feedback

File: CONTRIBUTING.adoc (hunk 1 above)
Comment (stevenhartley) [reply CONTRIBUTING.adoc]: fixed, thanks

File: CONTRIBUTING.adoc [hunk 4]
```diff
@@ -7,22 +7,22 @@ The key words "*MUST*", "*MUST NOT*", "*REQUIRED*", "*SHALL*", "*SHALL NOT*", "*
 == Getting Started
 
 === Eclipse Account
-First thing you need to do before contributing is to create an account on the Eclipse Foundation. You can create an account by visiting https://accounts.eclipse.org/user/register[here]. Once your account is created, you will need to sign the Eclipse Contributor Agreement (ECA) which can be found at http://www.eclipse.org/legal/ECA.php[here]. 
+First thing you need to do before contributing is to https://accounts.eclipse.org/user/register[create an account with the Eclipse Foundation]. Once your account is created, you will need to http://www.eclipse.org/legal/ECA.php[sign the Eclipse Contributor Agreement (ECA)].```
```
Comment (sophokles73) [inline CONTRIBUTING.adoc:10]: somehow three back-ticks (```) have snug in at the end of the line which do not belong there ...
